### PR TITLE
[FrameworkBundle] Add parameter types from Translation

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179 
| License       | MIT
| Doc PR        | N/A

This PR makes FrameworkBundle incompatible with Translation 4.4. Let's merge whenever we feel ready to do the bump.

### CODE REVIEW

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php [hunk 1]
```diff
@@ -117,13 +117,13 @@ public function addResource($format, $resource, $locale, $domain = null)
...
      */
-    protected function initializeCatalogue($locale)
+    protected function initializeCatalogue(string $locale)
     {
         $this->initialize();
         parent::initializeCatalogue($locale);
     }
 
-    protected function doLoadCatalogue($locale): void
+    protected function doLoadCatalogue(string $locale): void
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: shouldn't this be internal (in 4.4 then)
if yes, then let's not add this one and use an annotation instead, to keep compat with translation v4.4

Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: That wouldn't change a thing. If we want to maintain compatibility with Translation 4.4, we must not add any of the types I've added to this file. I'd simply leave this PR open until 5.0 has been branched out.

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: ok, then let's still add the missing `@internal` annotation to 4.3

File: src/Symfony/Component/Translation/Translator.php [hunk 2]
```diff
@@ -359,7 +359,7 @@ private function getCatalogueCachePath(string $locale)
     /**
      * @internal
      */
-    protected function doLoadCatalogue($locale): void
+    protected function doLoadCatalogue(string $locale): void
```
Comment (derrabus) [inline src/Symfony/Component/Translation/Translator.php]: Will be fixed with #33154

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**